### ISSUE TITLE
Heavies running out of power after having queue reset by low power and calling power_recharge

### ISSUE THREAD

**Issue by AdamSlay** [AdamSlay/luxai-bots#3]
There is some mismanagement of heavy power levels that needs to get rooted out. Hopefully a solution can be applied to lights as well.

**Comment by AdamSlay** [comment]
I think this has to do with the fact that mining_queue() queues up an amount of dig actions that would leave the unit under the threshold to trigger a low power state(100) which causes power_recharge() to be triggered. The problem is that when mining_queue only queues up one extra dig, so the next action in the queue is a move command back toward the home factory. Since that is queued, the tile that would take the unit closer to home is entered into next_positions, then when power_recharge(0 gets called, it tries to move_toward home_f and ends up moving to the side rather than toward home because the tile that it wants to move to is considered off_limits.

**Comment by AdamSlay** [comment]
Addressed in commit b9aba8bd98458b1bbd4d8147d3fa8098ec39829e

Basically I made a new Agent method that removes the old next_position from new_positions whenever an old queue gets interrupted by a new one. The bug still occurs when a friendly unit occupying the tile that the unit wants to move to. So, I added a condition to find_new_direction that prefers moving center if center is available. However, upon testing, this just resulted in units staying still for many steps. So, I fudged it with a conditional check that game_state.real_env_steps is an even number. That way, about half the time that a unit tries to find a new space, it will instead just wait a step before trying to find a space. Not really sure if that actually helps tbh
